strawberryfields/parameters.py

```diff
@@ -187,6 +190,17 @@ def do_evaluate(p):
         is_tf = (type(v).__module__.startswith('tensorflow') for v in vals)
         printer = 'tensorflow' if any(is_tf) else 'numpy'
         func = sympy.lambdify(atoms, p, printer)
+
+        if dtype is not None:
```

Comment:
With the boolean solution from my other comment, this could read as e.g. `if casting:` and then the value of `printer` could determine which casting should take place.